velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -28,6 +28,11 @@ template <typename TSum>
 struct SumCount {
   TSum sum{0};
   int64_t count{0};
+  virtual ~SumCount() {}
+};
+
+struct DecimalAccumulator : SumCount<int128_t> {
+  int64_t overflow{0};
```

Comment:
It feels like we are just extending the 128 bits to 128 + 64 bits?  What if this still overflows?  Maybe 128 + 128 is safer.

Reply:
So, the overflow would be the value of MSB of uint128_t addition. It can be 1 or 0. For us to exhaust 64-bit space, there must be INT64_MAX number of addition operations and each addition must overflow. For example, while adding
1 million values of 10<sup>38</sup> - 1, we get the following overflow and sum values:

```
overflow:   587747
Total sum: 1865291224403587248969154354545212864
```
Here, overflow is a factor that later gets multiplied by OVERFLOW_MULTIPLIER (1 << 127).

Reply:
Thanks @karteekmurthys I see your point.  My question is, if we add more than 2<sup>65</sup> rows of 10<sup>38</sup> - 1, we still get overflow.  Maybe that number of rows is not realistic.